src/rgw/rgw_data_sync.cc

```diff
@@ -3139,24 +3121,26 @@ int RGWBucketShardIncrementalSyncCR::operate()
         /* not waiting for child here */
       }
     }
-
     tn->unset_flag(RGW_SNS_FLAG_ACTIVE);
+
+    if (syncstopped) {
+      // transition back to StateInit in RGWRunBucketSyncCoroutine. if sync is
+      // still disabled, we'll delete the sync status object. otherwise we'll
+      // restart full sync to catch any changes that happened while sync was
+      // disabled
+      sync_info.state = rgw_bucket_shard_sync_info::StateInit;
```

Comment:
@cbodley what happens if we crash here before this becomes durable?

Reply:
because the for-loop stops at `entries_end`, which is the first instance of `OP_SYNCSTOP` if present, we'll never write an incremental sync marker past this entry. so after crash, we'll restart incremental sync at a previous marker, and hit this case again